src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -410,6 +468,94 @@ public function doSave($data, $form)
         return $this->redirectAfterSave($isNewRecord);
     }
 
+    /**
+     * Goes to the next record
+     * @param  array $data The form data
+     * @param  Form $form The Form object
+     * @return HTTPResponse
+     */
+    public function doNext($data, $form)
+    {
+        Controller::curr()->getResponse()->addHeader("X-Pjax", "Content");
+        $link = $this->getEditLink($this->getNextRecordID());
+        return Controller::curr()->redirect($link);
+    }
+
+    /**
+     * Goes to the previous record
+     * @param  array $data The form data
+     * @param  Form $form The Form object
+     * @return HTTPResponse
+     */
+    public function doPrev($data, $form)
+    {
+        Controller::curr()->getResponse()->addHeader("X-Pjax", "Content");
+        $link = $this->getEditLink($this->getPreviousRecordID());
+        return Controller::curr()->redirect($link);
+    }
+
+    /**
+     * Creates a new record. If you're already creating a new record,
+     * this forces the URL to change.
+     *
+     * @param  array $data The form data
+     * @param  Form $form The Form object
+     * @return HTTPResponse
+     */
+    public function doNew($data, $form)
+    {
+        return Controller::curr()->redirect($this->owner->Link('addnew'));
+    }
+
+    /**
+     * Gets the edit link for a record
+     *
+     * @param  int $id The ID of the record in the GridField
+     * @return string
+     */
+    public function getEditLink($id)
+    {
+        return Controller::join_links(
+            $this->owner->gridField->Link(),
+            "item",
+            $id,
+            // todo: use http header instead
+            '?gridState=' . urlencode($this->owner->gridField->getState(false)->Value())
```

Comment:
Is there ever a case for `gridState` being in the request body? If not, I would remove your check to `POST` above.